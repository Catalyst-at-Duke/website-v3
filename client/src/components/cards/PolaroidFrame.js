import React from "react";
import { colors, fonts } from "../../styles/theme.js";
import "./cards.css";
import * as ReactBootstrap from "react-bootstrap";

export default class PolaroidFrame extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hovered: false,
    };
  }
  render() {
    return (
      <div
        className="polaroid-outer"
        style={{ backgroundColor: colors.white }}
        onMouseLeave={() => this.setState({ hovered: false })}
        onMouseEnter={() => this.setState({ hovered: true })}
      >
        <div className="polaroid-inner">
          <img
            style={{
              backgroundColor: colors.black,
              width: "200px",
              height: "190px",
              objectFit: "cover",
            }}
            src={this.props.photo}
          />
        </div>
        <div style={{ fontSize: fonts.size.normal }}>{this.props.name}</div>
        <div
          style={{
            fontSize: fonts.size.normal,
            fontWeight: fonts.weights.light,
          }}
        >
          {this.props.position}
        </div>

        <div
          style={{
            position: "absolute",
            width: "230px",
            overflow: "auto",
            height: "280px",
            backgroundColor: colors.white,
            padding: "5px",
            transition: "opacity .2s",
            opacity: this.state.hovered ? "95%" : "0%",
          }}
        >
          {this.props.message}
        </div>
      </div>
    );
  }
}
