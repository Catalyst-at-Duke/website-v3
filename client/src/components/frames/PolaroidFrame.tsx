import React, { useState } from "react";
import { colors, fonts } from "../../styles/theme.js";
import "./style.css";
// import Tape from "../../images/tape.png";

export interface PublicProps {
  message: string;
  name: string;
  photo: string;
  position: string;
}

const PolaroidFrame = (props: PublicProps) => {
  const [isHovered, setIsHovered] = useState(false);

  const { message, name, photo, position } = props;
  return (
    <div>
      {/* <div style={{ background: `url(${Tape})` }}></div> */}
      <div
        className="polaroid-outer"
        style={{ backgroundColor: colors.white }}
        onMouseLeave={() => setIsHovered(false)}
        onMouseEnter={() => setIsHovered(true)}
      >
        <div className="polaroid-inner">
          <img
            alt="Catalyst member"
            style={{
              backgroundColor: colors.black,
              width: "200px",
              height: "190px",
              objectFit: "cover",
            }}
            src={photo}
          />
        </div>
        <div style={{ fontSize: fonts.size.normal }}>{name}</div>
        <div
          style={{
            fontSize: fonts.size.normal,
            fontWeight: "lighter",
          }}
        >
          {position}
        </div>

        <div
          style={{
            position: "absolute",
            width: "230px",
            overflow: "auto",
            height: "280px",
            backgroundColor: colors.white,
            padding: "5px",
            transition: "opacity .2s",
            opacity: isHovered ? "95%" : "0%",
          }}
        >
          {message}
        </div>
      </div>
    </div>
  );
};

export default PolaroidFrame;
