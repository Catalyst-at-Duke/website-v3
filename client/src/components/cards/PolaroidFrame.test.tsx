import React from "react";
import { shallow, configure } from "enzyme";
import PolaroidFrame, { PublicProps } from "./PolaroidFrame";

describe("PhoneFrame with no props", () => {
  const polaroid = shallow(
    <PolaroidFrame
      message={"testing"}
      name={"First Last"}
      photo={"img-src"}
      position={"supreme chairman"}
    />
  );
  it("should match the snapshot", () => {
    expect(polaroid.html()).toMatchSnapshot();
  });
});
