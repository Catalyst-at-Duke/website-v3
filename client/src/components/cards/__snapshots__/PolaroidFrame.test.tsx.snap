// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneFrame with no props should match the snapshot 1`] = `"<div><div class=\\"polaroid-outer\\" style=\\"background-color:#ffffff\\"><div class=\\"polaroid-inner\\"><img style=\\"background-color:#000;width:200px;height:190px;object-fit:cover\\" src=\\"img-src\\"/></div><div style=\\"font-size:1.125em\\">First Last</div><div style=\\"font-size:1.125em;font-weight:lighter\\">supreme chairman</div><div style=\\"position:absolute;width:230px;overflow:auto;height:280px;background-color:#ffffff;padding:5px;transition:opacity .2s;opacity:0%\\">testing</div></div></div>"`;
